### PR TITLE
MICAH - Add working paths, functional local search by tags, Datastore Integration

### PR DESCRIPTION
**Pull request by bzreinhardt**
Lots of different things (I need to get better at separating functions into different PRS)
- Integrates `DataStore` into the App so everything is managed through the `DataStore` API
- Changes the ID convention on Tags and People: before it was random uuids - now it is human readable: this will make everything easier going forward
- Adds logic around what should happen based off of which state the app is in
- Makes all the paths work:
`/search/label1+label2+label3` goes into search mode for all those labels
`/person/personID`   goes into person mode and lets you add tags

### CODE REVIEW

File: my-app/src/DataStore.jsx [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
 import tags from './ConstData/tags.js';
 
+export const uuid = foo => {
+  // Get a random uuid
+  // This is temporary until the database is hooked up
+  return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+  // eslint-disable-next-line
+    (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
+  )
+}
```
Comment (ytsheng) [inline my-app/src/DataStore.jsx]: delete.

**Comment by ytsheng** [comment]
Looks good overall; Refactor `getMatchingPersons`. :)

**Comment by ytsheng** [comment]
👍

**Comment by bzreinhardt** [comment]
From Sasha, @adriennetran, @micahcatlin, if you two feel good about `getMatchingPersons` then I am fine. As I was reading through this, it was confusing for me but Ben explained it and it made sense now.

File: my-app/src/DataStore.jsx [hunk 2]
```diff
@@ -1,8 +1,15 @@
-import React from 'react';
+
 //Remove these when we're done with constant data
 import persons from './ConstData/persons.js';
 import tags from './ConstData/tags.js';
 
+export const nameToId = name => {
+  return name.replace(/[^A-Z0-9]/ig, "_") + crypto.getRandomValues(new Uint8Array(1));
```
[the remark this answers is not in the document]
Comment (bzreinhardt) [reply my-app/src/DataStore.jsx]: That was the idea in my head, but apparently not in my hands 🤦‍♂️

File: my-app/src/Omnibox/Omnibox.jsx [hunk 3]
```diff
@@ -4,6 +4,25 @@ import Autosuggest from 'react-autosuggest';
 
 
 //const focusInputOnSuggestionClick = !isMobile.any;
+const uniqLabelFast = a => {
```
[the remark this answers is not in the document]
Comment (bzreinhardt) [reply my-app/src/Omnibox/Omnibox.jsx]: This function doesn't get a list of unique labels  - it gets a list of tags such that all the labels in them are unique.

File: my-app/src/Search/Search.jsx [hunk 4]
```diff
@@ -1,25 +1,62 @@
 import React, { Component } from 'react';
-import {
-  Route,
-} from 'react-router-dom'
 
-class Search extends Component {
-  constructor({match}) {
+const getSearchLabels =(searchString) => {
+  var labels = searchString.split("+");
+  var formattedLabels = [];
```
[the remark this answers is not in the document]
Comment (bzreinhardt) [reply my-app/src/Search/Search.jsx]: 👍

**Pull request merged**